### PR TITLE
Feature/favorite max notification TM-1618

### PR DESCRIPTION
**Pull request by scott062**
- Add system message for max favorites of 250 (for either ap or pv - so 500 total)
- Add test for 507 http error code response
- Conditional toast error notification based on 507 res

Depends on: https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/367

### CODE REVIEW

**Comment by scott062** [comment]
Note: deleted duplicate branch that I had unnecessarily branched off of Elizabeth's pagination PR

File: src/Constants/SystemMessages.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ export const DELETE_FAVORITE_SUCCESS = (pos, onToggle) => RemoveSuccess({ pos, o
 export const DELETE_FAVORITE_ERROR = () => "We're experiencing an error attempting to remove this position from your Favorites. Please try again.";
 export const ADD_FAVORITE_SUCCESS = pos => FavoriteSuccess({ pos });
 export const ADD_FAVORITE_ERROR = () => "We're experiencing an error attempting to add this position to your Favorites. Please try again.";
+export const ADD_FAVORITE_LIMIT_ERROR = 'You have reached the favorites limit of 250. Please remove a favorite and try again.';
```
Comment (mjoyce91) [inline src/Constants/SystemMessages.js]: Can we have the API own and send back the limit number? Then the front-end doesn't need to be a source of truth and would reduce any back and forth changes should we decide to change the limit.

**Comment by elizabeth-jimenez** [comment]
Discussed in Slack, just putting here for consistency: Custom Toast notification for AP vs PV. 

Ex: You have reached favorites limit of 250 for Open Positions. Please remove an Open Position favorite and try again.

File: src/actions/userProfile.js [hunk 2]
```diff
@@ -227,9 +227,18 @@ export function userProfileToggleFavoritePosition(id, remove, refreshFavorites =
...
+        let message = '';
+        if (remove) {
+          message = SystemMessages.DELETE_FAVORITE_ERROR();
+        } else if (response.status === 507 && isPV) {
+          message = SystemMessages.ADD_FAVORITE_LIMIT_ERROR_PV(limit);
+        } else if (response.status === 507) {
+          message = SystemMessages.ADD_FAVORITE_LIMIT_ERROR_AP(limit);
+        } else {
+          message = SystemMessages.ADD_FAVORITE_ERROR();
+        }
```
Comment (mjoyce91) [inline src/actions/userProfile.js:241]: 👍

File: src/actions/userProfile.js [hunk 3]
```diff
@@ -227,9 +227,18 @@ export function userProfileToggleFavoritePosition(id, remove, refreshFavorites =
...
         const title = SystemMessages.ERROR_FAVORITE_TITLE;
```
Comment (mjoyce91) [inline src/actions/userProfile.js:242]: Let's add a check for if `limit` is undefined, and if so, return the generic error.

```js
...
} else if (response.status === 507 && isPV && limit) {
...
} else if (response.status === 507 && limit) {
...
```

**Pull request merged**